docs: collections in the library (#75448)
metabase/metabase: docs/data-studio/library.md

- [Publish a table](#publishing-tables)
- [Create a metric](#metrics)
- [Create a SQL snippet or folder](#sql-snippets)
- [Create a subcollection or snippet folder](#library-organization)

## Library organization

![Library organization](./images/library-org.png)

Library is essentially a special collection. It has three special "subcollections":

- **Data** - for [published tables](#publishing-tables);
- **Metrics** - for [official metrics](#metrics);
- **Snippets** - for all the [SQL snippets](#sql-snippets) on your instance.

> These special collections are pre-defined - you can't get rid of them or rename them (but you can use [Library permissions](#library-permissions) to control who sees these collections).

Reviewer: ```suggestion
These special collections are predefined. You can't rename or archive them, (but you can use [ permissions](#library-permissions) to control who sees these collections).
```

> Each of these special collections can have further subcollections. For example, if your instance has a lot of published tables, you might want to break the **Library > Data** collection into "Sales", "Marketing", "Product" tables etc.

Reviewer: ```suggestion
Each of these special collections can have further subcollections. For example, if your Metabase has a lot of published tables, you might want to organize the **Library > Data** collection into "Sales", "Marketing", "Product" subcollections.
```

> People will see the Library structure in the navigation sidebar and in the data picker in the query builder:

Reviewer: ```suggestion
```
Self-evident.

Author: No it's not, because this is the first time in this doc we're mentioning the effect the Library has on the data picker.

Tables published to the Library remain available via the data browser as well.

To publish a table to the library:

1. Go to **Data Studio > Library**

### Managing tables

Once a table is published, you can view and manage its metadata, and more.

- Overview
- Fields
- [Segments](segments.md)

> - [Measures]

Author: ```suggestion
- [Measures](measures.md)
```

> ## Library permissions

Reviewer: This section should be in the collection permissions doc.

Library is essentially just a special collection. Metabase uses the standard [collection permissions](../permissions/collections.md) to determine who can view and edit items in the Library.

> ### Permissions to edit the Library

Reviewer: I see what you're doing, but---while fastidious---this section (and the one below it) overcomplicate the situation. We should do some of the mental work of simplifying this setup for people. 

To do so, we should probably stick to curate, view, no access, and explain what those mean. It'll make things easier to explain. And some kind of heuristic: library perms are for library viz and metric curation only.

For example (and this should be in the collection perms page, not the library page).

```
## Library permissions

Set permissions on the Library to determine which groups can view the Library, and which can add or remove metrics from the Library. 
```

Sample section:

```
## Curate permissions

Groups with curate permissions can do everything in View permissions, plus they can add or remove metrics from the Metrics collection.

The Admin and Data analyst groups always have Curate access to the Library.
```

And a note about what's different:

```
## The Data and Snippet collections

The Admin and Data Analyst groups are the _only_ groups that can publish tables to the Library's Data collection and its subcollections. Manage permissions to tables in the Data collection with data permissions like you would with any other table.

While snippets are automatically included in the Library's Snippet collection, you manage snippet permission via snippet folders.
```
With links, etc.

Author: "To do so, we should probably stick to curate, view, no access, and explain what those mean"

this is starting with a feature, not with a use case. A person thinking "i want to restrict people from seeing some tables" does not know what exactly what they need to do, much less what those kind of permissions are called in Metabase, that _why_ they're going to the docs. Docs exist to help people accomplish their goals, not walk through every piece of the interface

![Library collection permissions](./images/library-permissions.png)

> Admins and people in the Data Analyst group always have Curate access to the Library. Like with the Admin group, you can't change the Data Analyst's Curate access to the Library.

Reviewer: ```suggestion
Admins and people in the Data Analyst group always have Curate access to the Library.
```
"Always" is sufficient.

Author: Your line, not mine 😉
